### PR TITLE
[Dynamo] Fix several bugs & code refactor in RangeVariable

### PR DESCRIPTION
**Pull request by yanboliang**
Fix bug in [7k github models](https://github.com/pytorch/pytorch/issues/93427): https://github.com/jansel/pytorch-jit-paritybench/blob/master/generated/test_clovaai_stargan_v2.py
```
E       TypeError: 'list' object cannot be interpreted as an integer
E       
E       from user code:
E          File "/scratch/ybliang/work/repos/pytorch-jit-paritybench/generated/test_clovaai_stargan_v2.py", line 335, in forward
E           idx = torch.LongTensor(range(y.size(0)))
```

cc @mlazos @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/variables/lists.py [hunk 1]
```diff
@@ -89,42 +89,51 @@ def call_method(
...
+            (stop,) = items_to_map
+        elif len(items_to_map) == 2:
+            start, stop = items_to_map
+        elif len(items_to_map) == 3:
+            start, stop, step = items_to_map
+        else:
+            raise AssertionError()
+
+        assert stop is not None
+        super().__init__([start, stop, step], **kwargs)
```
Comment (yanboliang) [inline torch/_dynamo/variables/lists.py:108]: ```range``` is similar to ```slice```, we only need to store ```[start, stop, step]``` in ```self.items```. Moved the unpack logics to ```unpack_var_sequence```. This refactor fixed the reported bug and made API more concise/consistent.

File: test/dynamo/test_functions.py [hunk 2]
```diff
@@ -217,6 +217,17 @@ def test_slice5(a):
...
+    def test_range1(a):
+        return torch.tensor(range(a.size(0)))
+
+    @make_test
+    def test_range2(x, y):
+        r = x + y
+        for i in range(x.size(0) + 2):
+            r = r / y
+        return r
+
```
Comment (yanboliang) [inline test/dynamo/test_functions.py:230]: There are ```test_restore_range``` and ```test_restore_range_iter``` in ```test_subgraphs.py``` used to test ```range``` works well after graph break, so only add these two to cover this bug fix.

File: torch/_dynamo/variables/lists.py [hunk 3]
```diff
@@ -89,42 +89,51 @@ def call_method(
...
-            codegen.create_load_const(self.value.stop),
-            codegen.create_load_const(self.value.step),
-            create_instruction("CALL_FUNCTION", 3),
+            variables.ConstantVariable(x).add_options(self)
+            for x in self.as_python_constant()
         ]
 
+    def reconstruct(self, codegen):
+        assert "range" not in codegen.tx.f_globals
+        codegen.append_output(codegen.create_load_global("range", add=True))
```
Comment (jansel) [inline torch/_dynamo/variables/lists.py:127]: `range` could be shadowed by a user defined global, we should load this via one of the helpers.

**Review by jansel — CHANGES_REQUESTED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**